test/api/test_tools.py

```diff
@@ -1867,6 +1867,23 @@ def test_map_over_collection_structured_like(self):
             assert second_collection_level['collection_type'] == 'list'
             assert second_collection_level['elements'][0]['element_type'] == 'hda'
 
+    @skip_without_tool("__SPLIT_COLLECTION_BY_NUMBER__")
+    def test_split_collection_by_number(self):
+        with self.dataset_populator.test_history() as history_id:
+            hdca_id = self.dataset_collection_populator.create_list_in_history(history_id, contents=[("A", "A"), ("B", "B")]).json()['id']
+            self.dataset_populator.wait_for_history(history_id, assert_ok=True)
+            inputs = {
+                "input": {'values': [dict(src="hdca", id=hdca_id)]},
+                "how|how_split": "number_of_collections",
+                "how|num_collecton": 2
+            }
+            generated_collections = self._run("__FILTER_FROM_FILE__", history_id, inputs, assert_ok=True)['generated_collections']
```

Comment:
This doesn't look right to me, and that's where the API test is failing in https://jenkins.galaxyproject.org/job/docker-api-py3/246/testReport/junit/test.api.test_tools/ToolsTestCase/test_split_collection_by_number/

Reply:
Currently, I cannot pass the input integer to the tool API test:

Trying this should pass the integer in:
```
inputs = {
                "input": {'values': [dict(src="hdca", id=hdca_id)]},
                "how|how_split": "number_of_collections",
                "how|num_collecton": {'values': 2}
            }
            self._run("__SPLIT_COLLECTION_BY_NUMBER__", history_id, inputs, assert_ok=True)
```

Although I get an AssertionError: `Request status code (500) was not expected value 200. Body was {u'err_msg': u'Uncaught exception in exposed API method:', u'err_code': 0` due to:
```
  File "/home/CSCScience.ca/dhole/repos/galaxy/lib/galaxy/webapps/galaxy/api/tools.py", line 576, in _patch_library_inputs
    for index, value in enumerate(v['values']):
TypeError: 'int' object is not iterable
```

If I try to make it a list, it goes back to telling me that:
```
def _report_status_code_error(response, expected_status_code):
        try:
            body = response.json()
        except Exception:
            body = "INVALID JSON RESPONSE <%s>" % response.content
        assertion_message = ASSERT_FAIL_STATUS_CODE % (response.status_code, expected_status_code, body)
>       raise AssertionError(assertion_message)
E       AssertionError: Request status code (400) was not expected value 200. Body was {u'err_data': {u'how|num_collection': u"parameter 'num_collection': the attribute 'value' must be set for non optional parameters"}, u'err_msg': u"parameter 'num_collection': the attribute 'value' must be set for non optional parameters", u'err_code': 0}
```

Any suggestions? Or am I ok with just the wrapper tests that did work with the below changes

Reply:
Sorry, I should have mentioned that `'generated_collections'` is not a valid dictionary key return be `self._run`, you input parameters seem fine.

Reply:
Oh I'll try it the way that the filter from file tool has it. Although the way I've been currently testing has only `self.run`  and it's been giving me those errors

Reply:
It really should work though.
Try `generated_collections = self._run("__FILTER_FROM_FILE__", history_id, inputs, assert_ok=True)['output_collections']`, which is probably what you want.

Reply:
There's no implicit collections involved at all, these are only created if the input is "mapped over", i.e if you specify the input collection type to `list` but you give a `list:list` you'll get implicit collections.